msdsl/generator/verilog.py

```diff
@@ -360,8 +365,15 @@ def make_constant_mul_signal(self, expr: Product):
         # create the output signal
         output = Signal(name=next(self.namer), format_=expr.format_)
 
-        # call the special multiplication macro
-        self.macro_call('MUL_CONST_REAL', str(constant.value), signal.name, output.name)
+        # call the special multiplication macro, avoiding a synthesis corner-case:
+        # https://forums.xilinx.com/t5/Synthesis/Possible-synthesis-bug-casting-integer-to-real-in-a-function/td-p/1140910
+        if isinstance(constant.value, Integral):
+            # avoid a synthesis corner-case:
+            # https://forums.xilinx.com/t5/Synthesis/Possible-synthesis-bug-casting-integer-to-real-in-a-function/td-p/1140910
+            const_as_str = str(float(constant.value))
+        else:
+            const_as_str = str(constant.value)
```

Comment:
same as previous comment, this is related to a synthesis bug for integer constants when HARD_FLOAT is defined